File: javascript/challenges/post-graduion/stack/sumElements/stack.js
'use strict';
class Stack {
    constructor(){
        this.top = null;
    }
    push(item){
        this.top = new Node(item, this.top)
    }
    pop(){
        if(this.top === null){return 'stack is empty';}
        let temp = this.top.value;
        this.top = this.top.next;
        return temp;
    }
    isEmpty(){
        if(this.top === null){
            return true;
        } else {
            return false;
        }
    }
    peek(){
        if(!this.isEmpty()){
            return this.top.value;
        }
    }
    getMax(){
        if(this.isEmpty()){return null;}
        let top = this.top;
        let max = this.top.value;
        while(top !== null){
            if(max < top.value){
                max = top.value; 
            }
            top = top.next; 
        }
        return max;
    }
}
class Node{
    constructor(value, next=null){
        this.value = value;
        this.next = next;
    }
}
module.exports = Stack;

File: python/whiteboards/append_in_place/tests/stack.py
# create node class
class Node:
    def __init__(self, value=None, next=None):
        self.value = value
        self.next = next            

# create stack
class Stack:
    def __init__(self):
        self.top = None
    
    def isEmpty(self):
        if self.top == None:
            return True
        else:
            return False

    def peek(self):
        if self.isEmpty():
            raise Exception('cannot peek empty stack')
        else:
            return self.top.value

    def push(self, val):
        self.top = Node(val, self.top)
        return True
    
    def pop(self):
        if self.isEmpty():
            raise Exception('cannot pop empty stack')
        else:
            temp = self.top.value
            self.top = self.top.next
            return temp